File: AccuPay/Repositories/CategoryRepository.vb
Option Strict On

Imports System.Threading.Tasks
Imports Microsoft.EntityFrameworkCore

Namespace Global.AccuPay.Repository

    Public Class CategoryRepository

        Public Async Function GetLoanTypeId() As Task(Of Integer?)

            Return Await GetCategoryId("Loan Type")

        End Function

#Region "Private Functions"
        Private Async Function GetCategoryId(categoryName As String) As Task(Of Integer?)

            Using context = New PayrollContext()

                Dim category = Await context.Categories.
                                Where(Function(c) Nullable.Equals(c.OrganizationID, z_OrganizationID)).
                                FirstOrDefaultAsync

                Return category?.RowID

            End Using

        End Function
#End Region

    End Class

End Namespace

File: Accupay.Data/Repositories/CategoryRepository.cs
using AccuPay.Data.Entities;
using AccuPay.Utilities.Extensions;
using Microsoft.EntityFrameworkCore;
using System.Linq;
using System.Threading.Tasks;

namespace AccuPay.Data.Repositories
{
    public class CategoryRepository
    {
        private readonly PayrollContext _context;

        public CategoryRepository(PayrollContext context)
        {
            _context = context;
        }

        public async Task<Category> GetByNameAsync(int organizationId, string categoryName)
        {
            if (string.IsNullOrWhiteSpace(categoryName)) return null;

            var category = await _context.Categories.
                Where(c => c.OrganizationID == organizationId).
                Where(c => c.CategoryName.Trim().ToLower() ==
                            categoryName.ToTrimmedLowerCase()).
                FirstOrDefaultAsync();

            return category;
        }
    }
}